src/java.desktop/unix/classes/sun/awt/X11/XDragSourceContextPeer.java

```diff
@@ -392,6 +393,41 @@ private boolean updateSourceAction(int state) {
         return true;
     }
 
+    /**
+     * Our X11 code expects the drop target window to be a top level window
+     * and to have the XA_WM_STATE property.
+     * This is not true when performing drag and drop from XWayland
+     * to a native Wayland application.
+     * In this case XWayland creates a dummy window with only one property,
+     * XdndAware.
+     *
+     * @param window to test
+     * @return true if window has XdndAware property when running under Wayland
+     */
+    private static boolean isXWaylandDndAwareWindow(long window) {
+        if (Toolkit.getDefaultToolkit() instanceof SunToolkit sunToolkit
```

Comment:
So if it is NOT an instance of sunToolkit .. then you carry on to the code that gets the property.
Is that what you meant to happen ?

Reply:
Fixed.